xcodebuild.go

```diff
@@ -234,11 +241,19 @@ func createXcodebuildTestArgs(params models.XcodeBuildTestParamsModel, xcodeMajo
 		xcodebuildArgs = append(xcodebuildArgs, "GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=YES", "GCC_GENERATE_TEST_COVERAGE_FILES=YES")
 	}
 
-	if xcodeMajorVersion >= 13 && params.RetryTestsOnFailure {
+	switch params.TestRepetitionMode {
+	case untilFailure:
+		xcodebuildArgs = append(xcodebuildArgs, "-run-tests-until-failure")
+	case retryOnFailure:
 		xcodebuildArgs = append(xcodebuildArgs, "-retry-tests-on-failure")
+	case upUntilMaximumRepetitions, none:
+		break
+	default:
+		panic("Invalid test repetition mode: " + params.TestRepetitionMode)
+	}
 
-		// TODO(STEP-1054): Allow customization of `-test-iterations`.
-		xcodebuildArgs = append(xcodebuildArgs, "-test-iterations", "2")
+	if xcodeMajorVersion >= 13 && params.RetryTestsOnFailure {
+		xcodebuildArgs = append(xcodebuildArgs, "-retry-tests-on-failure", "-test-iterations", "2")
```

Comment:
Right now it could happen that both TestRepetitionMode and RetryTestsOnFailure is used and add duplicate or conficting options.

Reply:
I think I've answered this in this comment: https://github.com/bitrise-steplib/steps-xcode-test/pull/173#discussion_r667838514.

Reply:
There is an open question if to fail explicitly if using "Should retry failing tests" with Xcode 13.

Reply:
After discussing it online, we decided to fail explicitly if `RetryTestsOnFailure` is `true` in Xcode 13. Error message has been clarified to make this error actionable.